src/PollingBlockTracker.test.ts

```diff
@@ -0,0 +1,2068 @@
+ */
+async function withPollingBlockTracker(
+  callback: WithBlockTrackerCallback<PollingBlockTracker>,
+): Promise<void>;
+/* eslint-disable-next-line jsdoc/require-jsdoc */
+async function withPollingBlockTracker(
+  ...args: [...any[], WithBlockTrackerCallback<PollingBlockTracker>]
+): Promise<void> {
+  const callback: WithBlockTrackerCallback<PollingBlockTracker> = args.pop();
+  const options =
+    (args[0] as WithBlockTrackerOptions<PollingBlockTracker>) ?? {};
+  await withBlockTracker<PollingBlockTracker>(
+    PollingBlockTracker,
+    options,
+    callback,
+  );
+}
+
+describe('PollingBlockTracker', () => {
+  describe('constructor', () => {
+    it('should throw if given no options', () => {
+      expect(() => new PollingBlockTracker()).toThrow(
+        'PollingBlockTracker - no provider specified.',
+      );
+    });
+
+    it('should throw if given options but not given a provider', () => {
+      expect(() => new PollingBlockTracker({})).toThrow(
+        'PollingBlockTracker - no provider specified.',
+      );
+    });
+
+    it('should return a block tracker that is not running', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(({ blockTracker }) => {
+        expect(blockTracker.isRunning()).toBe(false);
+      });
+    });
+  });
+
+  describe('getLatestBlock', () => {
+    it('should start the block tracker immediately after being called', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(async ({ blockTracker }) => {
+        const promiseToGetLatestBlock = blockTracker.getLatestBlock();
+        expect(blockTracker.isRunning()).toBe(true);
+        // We have to wait for the promise to resolve after the assertion
+        // because by the time this promise resolves, the block tracker isn't
+        // running anymore
+        await promiseToGetLatestBlock;
+      });
+    });
+
+    it('should stop the block tracker automatically after its promise is fulfilled', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(async ({ blockTracker }) => {
+        await blockTracker.getLatestBlock();
+        expect(blockTracker.isRunning()).toBe(false);
+      });
+    });
+
+    it('should fetch the latest block number', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(
+        {
+          provider: {
+            stubs: [
+              {
+                methodName: 'eth_blockNumber',
+                response: {
+                  result: '0x0',
+                },
+              },
+            ],
+          },
+        },
+        async ({ blockTracker }) => {
+          const latestBlockNumber = await blockTracker.getLatestBlock();
+          expect(latestBlockNumber).toStrictEqual('0x0');
+        },
+      );
+    });
+
+    it('should cause the request to get the latest block to be made with `skipCache: true` if setSkipCacheFlag: true is given', async () => {
```

Comment:
```suggestion
    it('should get the latest block with `skipCache: true` if setSkipCacheFlag: true is given', async () => {
```

Reply:
Addressed in 130aabea3ad7cd9923828525a8e0338be93840d2 and commits/aaa6a794ae8a36813fb843f0ba6761dce26f3ee3.